samples/compiler_plugins/xnnpack/src/xnnpack/IR/XnnpackOps.td

```diff
@@ -45,11 +45,12 @@ def Xnnpack_FullyConnectedNcQd8F32Qc4wOp : Xnnpack_Op<"fully_connected_nc_qd8_f3
   let summary = "Fully connected layer with int8 input, int4 kernel, and f32 output";
   let arguments = (ins AnyRankedTensor:$input,
                        AnyRankedTensor:$kernel,
-                       BoolAttr:$transpose_rhs);
+                       BoolAttr:$transpose_rhs,
+                       IndexAttr:$kernel_id);
```

Comment:
Please add a note for kernel_id to say why we need this by describing the problem of using the kernel pointer address as the cache key.